hedera-node/src/main/java/com/hedera/services/ledger/accounts/staking/StakePeriodManager.java

```diff
@@ -82,8 +110,28 @@ public long effectivePeriod(final long stakePeriodStart) {
 		return stakePeriodStart;
 	}
 
+	private long stakePeriodStartUpdateFor(
+			final long curStakedId,
+			final long newStakedId,
+			final boolean rewarded
+	) {
+		// There's no reason to update stakedPeriodStart for an account not staking to
+		// a node; the value will never be used, since it cannot be eligible for a reward
+		if (newStakedId < 0) {
+			if (curStakedId >= 0) {
+				// We just started staking to a node
+				return currentStakePeriod();
+			} else {
+				// If we were just rewarded, stake period start is yesterday; otherwise, unchanged
+				return !rewarded ? -1 : currentStakePeriod() - 1;
```

Comment:
If we have just rewarded we should not update `stakePeriodStart` right, since we never reward again ?

If we are not rewarded should n't the stakePeriodStart return `currentStakePeriod()` instead of -1 ?

Reply:
Added javadoc to clarify; I think there was a missing condition here though, ty!